### PR TITLE
Adds a way Hosts can add error without extending Engine.

### PR DESCRIPTION
**Pull request by jas-valgotar**
Hosts can now add custom host specific authoring error without extending the Engine class.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Texl/PostCheckErrorHandler.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx
+{
+    public interface IPostCheckErrorHandler
+    {
+        public IEnumerable<ExpressionError> Process(TexlNode root);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/PostCheckErrorHandler.cs:11]: >TexlNode [](http://example.com/codeflow?start=52&length=8)

CheckResult? so they can access binding

File: src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs [hunk 2]
```diff
@@ -89,7 +89,12 @@ public Engine(PowerFxConfig powerFxConfig)
         /// <summary>
         /// List of transforms to apply to an IR. 
         /// </summary>
-        internal readonly List<Core.IR.IRTransform> IRTransformList = new List<Core.IR.IRTransform>();
+        internal readonly List<Core.IR.IRTransform> IRTransformList = new List<Core.IR.IRTransform>();
+        
+        /// <summary>
+        /// List of error processor that will add host specific custom errors at the end of the check.
+        /// </summary>
+        internal readonly IList<IPostCheckErrorHandler> PostCheckErrorHandlers = new List<IPostCheckErrorHandler>();
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs:97]: >internal [](http://example.com/codeflow?start=8&length=8)

public property?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs (hunk 2 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs]: updated

File: src/libraries/Microsoft.PowerFx.Core/Texl/PostCheckErrorHandler.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Texl/PostCheckErrorHandler.cs]: updated

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**